src/elliptic/curves/curve_ristretto.rs

```diff
@@ -6,593 +6,347 @@
-                bytes_array_32.copy_from_slice(&bytes_slice);
-                let r_point: PK = CompressedRistretto::from_slice(&bytes_array_32);
-                let r_point_compress = r_point.decompress();
-                match r_point_compress {
-                    Some(x) => {
-                        let new_point = RistrettoCurvPoint {
-                            purpose: "random",
-                            ge: x.compress(),
-                        };
-                        Ok(new_point)
-                    }
-                    None => Err(InvalidPublicKey),
-                }
-            }
-
-            _ => {
-                let bytes_slice = &bytes_vec[0..32];
-                bytes_array_32.copy_from_slice(&bytes_slice);
-                let r_point: PK = CompressedRistretto::from_slice(&bytes_array_32);
-                let r_point_compress = r_point.decompress();
-                match r_point_compress {
-                    Some(x) => {
-                        let new_point = RistrettoCurvPoint {
-                            purpose: "random",
-                            ge: x.compress(),
-                        };
-                        Ok(new_point)
-                    }
-                    None => Err(InvalidPublicKey),
-                }
-            }
-        }
+    fn generator() -> &'static RistrettoPoint {
+        &GENERATOR
     }
 
-    fn pk_to_key_slice(&self) -> Vec<u8> {
-        let result = self.ge.to_bytes();
-        result.to_vec()
+    fn base_point2() -> &'static RistrettoPoint {
+        &BASE_POINT2
     }
 
-    fn scalar_mul(&self, fe: &SK) -> RistrettoCurvPoint {
-        let skpk = fe * (self.ge.decompress().unwrap());
-        RistrettoCurvPoint {
-            purpose: "scalar_point_mul",
-            ge: skpk.compress(),
+    fn from_coords(x: &BigInt, y: &BigInt) -> Result<RistrettoPoint, NotOnCurve> {
+        let mut y_bytes = y.to_bytes_array::<32>().ok_or(NotOnCurve)?;
+        if x != &BigInt::from_bytes(&Sha256::digest(&y_bytes)) {
+            return Err(NotOnCurve);
         }
-    }
 
-    fn add_point(&self, other: &PK) -> RistrettoCurvPoint {
-        let pkpk = self.ge.decompress().unwrap() + other.decompress().unwrap();
-        RistrettoCurvPoint {
-            purpose: "combine",
-            ge: pkpk.compress(),
-        }
-    }
+        y_bytes.reverse();
+        let compressed = CompressedRistretto::from_slice(&y_bytes);
 
-    fn sub_point(&self, other: &PK) -> RistrettoCurvPoint {
-        let pkpk = self.ge.decompress().unwrap() - other.decompress().unwrap();
-        RistrettoCurvPoint {
-            purpose: "sub",
-            ge: pkpk.compress(),
-        }
+        Ok(RistrettoPoint {
+            purpose: "from_coords",
+            ge: compressed.decompress().ok_or(NotOnCurve)?,
+        })
     }
 
-    fn from_coor(_x: &BigInt, _y: &BigInt) -> RistrettoCurvPoint {
-        unimplemented!();
+    fn x_coord(&self) -> Option<BigInt> {
```

Comment:
`x_coord` and `from_coord` should be kept unimplemented . The discussion you point to  is not good enough evidence to take the hash of `y`

Reply:
Is it possible to define function similar to [xrecover] from ed25519 but for ristretto curve?

[xrecover]: https://github.com/ZenGo-X/curv/blob/34873cd80819a9ec75992a6b84da8507e14b6210/src/elliptic/curves/ed25519.rs#L467

Reply:
I changed ristretto curve to return x=None instead of x=hash(y), and from_coords to always returns `Err(NotOnCurve)`